packages/ui/components/overlays/test/OverlayController.test.js

```diff
@@ -606,6 +606,31 @@ describe('OverlayController', () => {
         await aTimeout(0);
         expect(ctrl.isShown).to.be.true;
       });
+
+      it('hides only one overlay when [escape] is pressed', async () => {
+        const ctrl1 = new OverlayController({
+          ...withGlobalTestConfig(),
+          hidesOnEsc: true,
+        });
+        const ctrl2 = new OverlayController({
+          ...withGlobalTestConfig(),
+          hidesOnEsc: true,
+        });
+
+        await ctrl1.show();
+        await ctrl2.show();
+        expect(ctrl1.manager.shownList.length).to.equal(2);
+        expect(ctrl1._blocksEscKeyHandler).to.be.true;
+        expect(ctrl2._blocksEscKeyHandler).to.be.false;
+
+        ctrl2.contentNode.dispatchEvent(new KeyboardEvent('keyup', { key: 'Escape' }));
+        await aTimeout(0);
+        expect(ctrl2.isShown).to.be.false;
+        expect(ctrl1.isShown).to.be.true;
```

Comment:
Does this test fail when `_blocksEscKeyHandler` is left out of the equation?

I would expect the keyup event to never reach ctrl1.contentNode, as ctrl2.contentNode and ctrl1.contentNode are siblings.

The test should probably create a setup where ctrl2.contentNode is nested inside ctrl1.contentNode